python/tests/test_highlevel.py

```diff
@@ -55,214 +54,6 @@
-            ancestral_state="0",
-            metadata=json.dumps({"index": j}).encode(),
-        )
-        tables.mutations.add_row(
-            site=j,
-            derived_state="1",
-            node=nodes[j % len(nodes)],
-            metadata=json.dumps({"index": j}).encode(),
-        )
-
-
-def get_table_collection_copy(tables, sequence_length):
-    """
-    Returns a copy of the specified table collection with the specified
-    sequence length.
-    """
-    table_dict = tables.asdict()
-    table_dict["sequence_length"] = sequence_length
-    return tskit.TableCollection.fromdict(table_dict)
-
-
-def insert_gap(ts, position, length):
-    """
-    Inserts a gap of the specified size into the specified tree sequence.
-    This involves: (1) breaking all edges that intersect with this point;
-    and (2) shifting all coordinates greater than this value up by the
-    gap length.
-    """
-    new_edges = []
-    for e in ts.edges():
-        if e.left < position < e.right:
-            new_edges.append([e.left, position, e.parent, e.child])
-            new_edges.append([position, e.right, e.parent, e.child])
-        else:
-            new_edges.append([e.left, e.right, e.parent, e.child])
-
-    # Now shift up all coordinates.
-    for e in new_edges:
-        # Left coordinates == position get shifted
-        if e[0] >= position:
-            e[0] += length
-        # Right coordinates == position do not get shifted
-        if e[1] > position:
-            e[1] += length
-    tables = ts.dump_tables()
-    L = ts.sequence_length + length
-    tables = get_table_collection_copy(tables, L)
-    tables.edges.clear()
-    tables.sites.clear()
-    tables.mutations.clear()
-    for left, right, parent, child in new_edges:
-        tables.edges.add_row(left, right, parent, child)
-    tables.sort()
-    # Throw in a bunch of mutations over the whole sequence on the samples.
-    insert_uniform_mutations(tables, 100, list(ts.samples()))
-    return tables.tree_sequence()
-
-
-def get_gap_examples():
-    """
-    Returns example tree sequences that contain gaps within the list of
-    edges.
-    """
-    ts = msprime.simulate(20, random_seed=56, recombination_rate=1)
-
-    assert ts.num_trees > 1
-
-    gap = 0.0125
-    for x in [0, 0.1, 0.5, 0.75]:
-        ts = insert_gap(ts, x, gap)
-        found = False
-        for t in ts.trees():
-            if t.interval[0] == x:
-                assert t.interval[1] == x + gap
-                assert len(t.parent_dict) == 0
-                found = True
-        assert found
-        yield ts
-    # Give an example with a gap at the end.
-    ts = msprime.simulate(10, random_seed=5, recombination_rate=1)
-    tables = get_table_collection_copy(ts.dump_tables(), 2)
-    tables.sites.clear()
-    tables.mutations.clear()
-    insert_uniform_mutations(tables, 100, list(ts.samples()))
-    yield tables.tree_sequence()
-
-
-def get_internal_samples_examples():
```

Comment:
I don't think we want to promote this code up to utils.py. The pattern of yielding a bunch of different examples isn't a great one as it leads to long-running tests where it's hard to distinguish which test has failed. I would prefer to leave this module as-is, and continue to factor tests out of it and into more appropriate places, rather than have this deprecated testing pattern diffuse out into other modules.

Creating a ts with a non-sample leaf is pretty easy:
```python
tables = msprime.simulate(n, random_seed=1).dump_tables()
root = len(tables.nodes) - 1
leaf = tables.nodes.add_row(flags=0, time=0)
tables.edges.add_row(0, 1, root, leaf)
tables.sort()
ts = tables.tree_sequence()
```

Reply:
Ah, OK. I do think we want a nice range of trees & tree sequences that we call upon from different test functions, rather than writing them all out again (e.g. the example above doesn't test internal samples, which would have been another good check). But this particular refactoring seems not to be the right way to go about it.